blaze/utils.py

```diff
@@ -187,6 +192,11 @@ def json_dumps(ds):
     return {'__!datashape': str(ds)}
 
 
+@dispatch(types.BuiltinFunctionType)
```

Comment:
More from the [docs](https://docs.python.org/2/library/types.html):
```
This module defines names for some object types that are used by the standard Python interpreter, but not for the types defined by various extension modules.
```

I now understand your comment better 👍 
But I believe this will be safe against functions compiled in extensions as per the docs and per some quick testing against `np.sum`